packages/afk_nostr_sdk/src/hooks/useRepost.ts

```diff
@@ -0,0 +1,33 @@
+import { NDKEvent, NDKKind } from '@nostr-dev-kit/ndk';
+import { useMutation, useQueryClient } from '@tanstack/react-query';
+import { useNostrContext } from '../context/NostrContext';
+
+export type UseRepostOptions = {
+  event?: NDKEvent;
+};
+
+export const useRepost = (options?: UseRepostOptions) => {
+  const { ndk } = useNostrContext();
+  const queryClient = useQueryClient();
+  return useMutation({
+    mutationFn: async () => {
+      if (!options?.event) {
+        throw new Error('No event provided for repost');
+      }
+      const repostEvent = new NDKEvent(ndk);
+      repostEvent.kind = NDKKind.Repost;
+      repostEvent.content = JSON.stringify(options.event.rawEvent());
+      repostEvent.tags = [
+        ['e', options.event.id, options.event.relay?.url || ''],
+        ['p', options.event.pubkey],
+      ];
+      await repostEvent.sign();
+      await ndk.publish(repostEvent);
```

Comment:
The ndk.publish() is deprecated.

Can you used the repostEvent?.publish() as the others hooks? That's sign the event and publish it at the same time